src/dagmc/dagmcmetadata.cpp

```diff
@@ -0,0 +1,591 @@
+      // to second delimiter
+      // by being here we already know he property exists
+      // being found upto the first delimiter
+      if (delimiters.size() > 1) {
+        for (int j = 0 ; j < properties.size() ; j++) {
+          size_t npos = 0, first = npos;
+          npos = properties[j].find(delimiters[1]);
+          // extract from the match - which is either first
+          // match or .length()
+          properties[j] = properties[j].substr(0, npos);
+        }
+      }
+    } else {
+      properties.push_back("");
+    }
+
+    if (properties.size() > 1)
+      if (remove_duplicates)
+        properties = remove_duplicate_properties(properties);
+    // assign the map value
+    prop_map[entity] = properties;
+  }
+
+  return prop_map;
+}
+
+// remove duplicate properties from the vector of properties
+std::vector<std::string> dagmcMetaData::remove_duplicate_properties(std::vector<std::string> properties) {
+  // remove duplicates
+  std::set<std::string> properties_set; // set of properties (unique)
+
+  std::vector<std::string>::iterator it;
+  // loop over all properties and insert them into a set
+
+  for (it = properties.begin() ; it != properties.end() ; ++it) {
+    properties_set.insert(*it);
+  }
+
+  // due to dagmc parse group names, the property, and value are two seperate
+  // entries in array, i.e a tag like bob:charlie/bob, will return as charlie and charlie/bob
+  // so we need to search each item for its more information rich partner and remove the
+  // degenerate item(s) - should probably be fixed upstream eventually
+
+  properties_set = set_remove_rich(properties_set);
+
+  std::set<std::string>::iterator iter;
+
+  std::vector<std::string> new_properties;
+  // resize the array
+  new_properties.resize(properties_set.size());
+
+  // turn set back into vector
+  std::copy(properties_set.begin(), properties_set.end(), new_properties.begin());
+
+  return new_properties;
+}
+
+// from a given set remove any matches if they are found in order to keep the
+// the information rich version. ie. if we find both neutron and neutron/1.0 keep
+// the second one
+std::set<std::string> dagmcMetaData::set_remove_rich(std::set<std::string> properties_set) {
+  std::set<std::string> new_set = properties_set;
+
+  std::vector<std::set<std::string>::const_iterator> matches;
+  std::set<std::string>::iterator set_it, toofar, it;
+
+  std::set<std::string> to_delete;
+  // loop over all elements in the set
+  it = new_set.begin();
+  while (it != new_set.end()) {
+    // loop over the set trying to find similar names
+    for (set_it = new_set.begin(), toofar = new_set.end(); set_it != toofar; ++set_it)
+      if ((*set_it).find(*it) != std::string::npos && (*set_it != *it)) {
+        matches.push_back(it);
+        matches.push_back(set_it);
+      }
+
+    // if there were more than 2 matches
+    if (matches.size() > 1) {
+      int smallest = 0;
+      int len = 1e7;
```

Comment:
Best Practices: If you are looking to initialize `len` with the maximum possible value of `int`, recommend using the `INT_MAX` constant.  This is more readable and also is compiler specific based on the bit-width of `int`.